lib/data/school_mass_importer.rb

```diff
@@ -1,9 +1,10 @@
-require 'breasal'
-require 'activerecord-import'
+require_relative 'school_data_helpers'
 
 class SchoolMassImporter
```

Comment:
This doesn't really relate to these changes, but I've been looking through the existing import in general with a view to the much bigger data set it will be importing, and it run nightly

```ruby
def import
    data = edubase_data
        .reject { |urn, _| urn.in?(existing_urns) }
        .map { |_, row| build_school(row) }
        .compact

    Bookings::School.transaction do
      puts "importing schools..."
      Bookings::School.import(data)

      puts "setting up phases..."
      edubase_data.each.with_index do |(urn, row), i|
        if (i % 1000).zero?
          print '.'
        end

        school = Bookings::School.find_by(urn: urn)

        next unless school.present?

        assign_phases(school, row['PhaseOfEducation (code)'])
      end

      puts "done..."
    end
end
```

This is doing a `find_by` for every row in the data set, irrespective of whether we've just imported it - this will be incurred on a nightly basis once its scheduled so its better if we can optimise it.

the `edubase_data.each.with_index do |(urn, row), i|` needs to have rejected the existing URNs, it can also do the find_by in slices?

eg

```ruby
data = edubase_data.reject { |urn, _| urn.in?(existing_urns) }
puts "setting up phases..."

data.each_slice(BATCH_SIZE) do |batch|
  puts "."
  
  indexed_rows = Hash[batch]

  Bookings::School.where(urn: batch.map(&:first)).each do |school|
     row = indexed_row[school.urn]

     assign_phases(school, row['PhaseOfEducation (code)'])
  end  
end
```

The above is a bit muddled, the `each_slice` turns the Hash into an Array of kv pairs, which it then converts back. But it gives a broad idea on operating on the data in a batched format?

Reply:
Yep, I missed this first time - the original intention was to run the phases step for 'new' schools. I've addressed this